### PR TITLE
Improve Credentials hierarchy

### PR DESCRIPTION
**Pull request by mackdk**
## What does this PR change?

This PR refactors the Credentials class by introducing a hierarchy based on the credential type.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [X] **DONE**

## Test coverage
- Test updated, already covered.

- [X] **DONE**

## Links

Issue: https://github.com/SUSE/spacewalk/issues/22014
Ports: https://github.com/SUSE/spacewalk/pull/23066

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java [hunk 1]
```diff
@@ -200,6 +200,17 @@ public static ProductName findOrCreateProductName(String name) {
...
     }
 
+    /**
+     * Converts the specified network url to a file system url
+     * @param urlString the url
+     * @param name the name of the repo
+     * @return the file system URI
+     */
+    public static URI urlToFSPath(String urlString, String name) {
+        String sccDataPath = Config.get().getString(ContentSyncManager.RESOURCE_PATH, null);
```
Comment (mcalmer) [inline java/code/src/com/redhat/rhn/manager/content/MgrSyncUtils.java:210]: ConfigDefaults got a new method "getOfflineMirrorDir()". Why is this not used here?

**Review by mcalmer — CHANGES_REQUESTED** [review]
some small nitpicks

**Review by srbarrios — APPROVED** [review]
I did not review the whole PR.
But I'm fine with the need of restart taskomatic, after building the code, in our GH validation.

**Review by mcalmer — APPROVED** [review]

**Review by parlt91 — APPROVED** [review]
Frontend changes look good

**Review by vzhestkov — APPROVED** [review]
At least from the python point of view it makes things more simple. So I like it :+1:

**Pull request merged**